config/initializers/edition_services.rb

```diff
@@ -15,6 +15,14 @@
         .new(attachment.attachment_data)
         .update!
     end
+    Attachment.where(attachable_id: edition.id).find_each do |attachment|
+      next unless attachment.attachment_data
+      AttachmentData.where(replaced_by_id: attachment.attachment_data.id).find_each do |attachment_data|
```

Comment:
This nested query makes me uneasy.

Reply:
I think you should be able to do something like:

```rb
AttachmentData.joins(:attachment).where(attachment: { attachable_id: edition.id }, replaced_by_id: attachment.attachment_data.id).find_each do |attachment_data|
  ...
```